diff --git a/node_modules/ember-power-select/addon/components/power-select.ts b/node_modules/ember-power-select/addon/components/power-select.ts
index 1111111..2222222 100644
--- a/node_modules/ember-power-select/addon/components/power-select.ts
+++ b/node_modules/ember-power-select/addon/components/power-select.ts
@@ -574,8 +574,7 @@ export default class PowerSelect extends Component<PowerSelectArgs> {
   }
 
   // Tasks
-  @restartableTask
-  *triggerTypingTask(this: PowerSelect, e: KeyboardEvent) {
+  triggerTypingTask = restartableTask(async (e: KeyboardEvent) => {
     // In general, a user doing this interaction means to have a different result.
     let searchStartOffset = 1;
     let repeatingChar = this._repeatingChar;
@@ -624,10 +623,10 @@ export default class PowerSelect extends Component<PowerSelectArgs> {
         this.storedAPI.actions.select(match, e);
       }
     }
-    yield timeout(1000);
+    await timeout(1000);
     this._expirableSearchText = '';
     this._repeatingChar = '';
-  }
+  });
 }
 
 function getOptionMatcher(matcher: MatcherFn, defaultMatcher: MatcherFn, searchField: string | undefined): MatcherFn {
