diff --git a/node_modules/ember-toggle/addon/components/x-toggle/template.hbs b/node_modules/ember-toggle/addon/components/x-toggle/template.hbs
index 1111111..2222222 100644
--- a/node_modules/ember-toggle/addon/components/x-toggle/template.hbs
+++ b/node_modules/ember-toggle/addon/components/x-toggle/template.hbs
@@ -1,4 +1,4 @@
-{{#if hasBlock}}
+{{#if (has-block)}}
   {{yield
     (hash
       switch=(component 'x-toggle-switch'
diff --git a/node_modules/ember-toggle/addon/components/x-toggle-label/template.hbs b/node_modules/ember-toggle/addon/components/x-toggle-label/template.hbs
index 3333333..4444444 100644
--- a/node_modules/ember-toggle/addon/components/x-toggle-label/template.hbs
+++ b/node_modules/ember-toggle/addon/components/x-toggle-label/template.hbs
@@ -5,7 +5,7 @@
     onclick={{action 'clickLabel'}}
     ...attributes
   > 
-    {{#if hasBlock}}
+    {{#if (has-block)}}
       {{yield this.label}}
     {{else}}
       {{this.label}}
